deploy.sh

```diff
@@ -118,18 +118,78 @@ for proj in $projects; do # bash only
   echo "||| $0: $proj"
   echo "|||"
 
+  # Function to fetch secrets from Google Cloud Secret Manager (uses same name for env var and secret)
+  fetch_secrets_to_env() {
+    # Check if gcloud is available
+    if ! command -v gcloud &> /dev/null; then
+      echo "[ERROR] gcloud command not found, skipping secret fetching"
+      return 1
+    fi
+
+    echo "Fetching secrets from Google Cloud Secret Manager for environment: $env..."
+
+    # Define the list of secrets to fetch - names will be identical in env and Secret Manager
+    local secrets=(
+      "STATSIG_CLIENT_KEY"
+      "STATSIG_SERVER_KEY"
+    )
+
+    # Add environment-specific Sentry token
+    local sentry_token_name
+    case "$env" in
+      "prod")
+        sentry_token_name="SENTRY_AUTH_TOKEN_PROD"
+        ;;
+      "staging")
+        sentry_token_name="SENTRY_AUTH_TOKEN_STAGING"
+        ;;
+      "local")
+        echo "[INFO] Skipping auth token fetching for local environment"
+        ;;
+      *)
+        echo "[ERROR] Unknown environment '$env'"
+        exit 1
+    esac
+
+    # Add Sentry token to secrets list if we're in prod/staging
+    if [ -n "$sentry_token_name" ]; then
+      secrets+=("$sentry_token_name")
+    fi
+
+    # Fetch each secret and export it to environment
+    for secret_name in "${secrets[@]}"; do
+      echo "  Fetching secret $secret_name..."
+
+      # Fetch the secret value
+      local value
+      value=$(gcloud secrets versions access latest --secret="$secret_name" --project=sales-engineering-sf 2>/dev/null)
+
+      if [ $? -ne 0 ]; then
+        echo "[WARN] Failed to fetch secret '$secret_name', skipping"
+      else
```

Comment:
Let fail early so it's clear that required secret it missing and we don't have to scratch our heads later why something is not working

```
>&2 echo "[ERROR] Failed to fetch secret '$secret_name'"
exit 1
```